### PR TITLE
Add the SSO button and its code to UI

### PR DESCRIPTION
**Pull request by Minivera**
Work is related to manifoldco/engineering#8691

## Reason for change
The SSO button is a data component that will trigger a POST sso request on connector and return the redirect_url necessary for properly logging into the resource's product dashboard.

It also adds a `manifold-resource-sso` wrapper components that will automatically inject some of the attributes into the button.

## Testing
Test with storybook

### CODE REVIEW

File: docs/docs/data/manifold-data-sso-button.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+`slot` can be attached to any HTML or JSX element. To read more about slots, see [Stencil’s Documentation][stencil-slot]
+
+## Events
+
+For validation, error, and success messages, it will emit custom events.
+
+```js
+document.addEventListener('manifold-ssoButton-click', ({ detail: { resourceLabel } }) =>
+  console.info(`⌛ Sooing into ${resourceLabel} …`)
```
Comment (seriousben) [inline docs/docs/data/manifold-data-sso-button.md]: Typo here. Should be SSOing.

File: src/components/manifold-resource-product/manifold-resource-product.tsx [hunk 2]
```diff
@@ -1,35 +1,38 @@
...
+    return this.data && this.data.product ? (
+      <manifold-service-view
+        name={this.data.product.name}
+        description={this.data.product.tagline}
+        label={this.data.product.label}
+        logo={this.data.product.logo_url}
+        productId={this.data.product.id}
+        asCard={this.asCard}
+      >
+        <manifold-forward-slot slot="cta">
```
Comment (batgirl) [inline src/components/manifold-resource-product/manifold-resource-product.tsx]: Why is it called `manifold-forward-slot` and not `manifold-slot`?

Comment (batgirl) [reply src/components/manifold-resource-product/manifold-resource-product.tsx]: I guess my question is only relevant if those components are client-facing :P

**Comment by batgirl** [comment]
I also see the `manifold-resource-list.e2e.ts` test failing on `renders the no resource state if no resources are found`.

File: docs/docs/components/manifold-service-card.md [hunk 3]
```diff
@@ -21,6 +21,20 @@ Compact view of a Manifold Product. Available as a self-fetching component or a
 ></manifold-service-card-view>
 ```
 
+## CTA
+
+You can pass in your own CTA content in the right of the component
+by passing in any element with `slot="cta"` as an attribute. [Read more about
+slots][slot].
```
[the remark this answers is not in the document]
Comment (Minivera) [reply docs/docs/components/manifold-service-card.md]: Yes! It seems to not be a link everywhere 🤔

File: docs/docs/data/manifold-data-sso-button.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+```js
+document.addEventListener('manifold-ssoButton-click', ({ detail: { resourceLabel } }) =>
+  console.info(`⌛ Ssoing into ${resourceLabel} …`)
+);
+document.addEventListener(
+  'manifold-ssoButton-success',
+  ({ detail: { resourceLabel, redirectUrl } }) => window.location = redirectUrl
+);
+document.addEventListener('manifold-ssoButton-error', ({ detail }) => console.log(detail));
+// {type: "not_found", message: "no_found: Resource not found", resourceid: "1234", resourceLabel: "my-resource"}
```
[the remark this answers is not in the document]
Comment (Minivera) [reply docs/docs/data/manifold-data-sso-button.md]: Yep, will fix

File: docs/docs/resource-details.md [hunk 5]
```diff
@@ -127,6 +130,41 @@ document.addEventListener('manifold-renameButton-invalid', ({ detail }) => conso
 | `manifold-renameButton-error`   |   `message`, `resourceId`, `resourceLabel`, `newLabel`   | Erred rename, along with information on what went wrong.                                                                    |
 | `manifold-renameButton-error`   |   `message`, `resourceId`, `resourceLabel`, `newLabel`   | Invalid renaming, along with information on what went wrong. 
 
+## The resource sso Button
```
[the remark this answers is not in the document]
Comment (Minivera) [reply docs/docs/resource-details.md]: There are none, the docs page "are" the preview

File: src/components/manifold-resource-product/manifold-resource-product.tsx (hunk 2 above)
Comment (Minivera) [reply src/components/manifold-resource-product/manifold-resource-product.tsx]: These are not, it's a helper sam built to work around slots not working in nested web components.

**Comment by Minivera** [comment]
That test fails on local only for some reason, you should be able to ignore it. I'll look at it when I get time.

**Comment by batgirl** [comment]
Awesome thank you for the follow up Guillaume!!

> In Storybook it's currently very hard to see the SSO flow as I expect it to be used - the in person demo today helped me understand how the CTA and the SSO button relate to each other. But I do see the SSO button making a (failing) call to `https://api.connector.manifold.co/v1/sso`! So if that's the expected behavior then it looks good to me.

Do you think this is okay to approve given this^ then?

**Comment by Minivera** [comment]
Where were you trying the sso button? In the docs?

**Pull request merged**